### PR TITLE
Add documentation for cirq google's support for google internal gates

### PR DESCRIPTION
**Pull request by NoureldinYosri**

### CODE REVIEW

File: docs/google/pyle_gates.ipynb [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "coupler = cirq_google.InternalGate(\n",
+    "    gate_module='pyle.cirqtools.pyle_gates',  # Module of class. \n",
```
Comment (dstrain115) [inline docs/google/pyle_gates.ipynb:128]: Let's remove the actual package name from public docs.  If we want a real example, let's add it internally.
Also, let's remove references to internal project name 'pyle' as much as possible.

Comment (NoureldinYosri) [reply docs/google/pyle_gates.ipynb]: done

File: docs/google/internal_gates.ipynb [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "cirq_google.InternalGate(gate_name=\"GATE_NAMPE\", gate_module=\"GATE_MODULE\", num_qubits=2, )"
```
Comment (dstrain115) [inline docs/google/internal_gates.ipynb:107]: typo NAMPE to NAME.

**Review by dstrain115 — APPROVED** [review]

**Comment by dstrain115** [comment]
Do we need to link this doc via any TOC anywhere, cc @wcourtney ?

**Comment by NoureldinYosri** [comment]
@dstrain115  I don't think any of the cirq-google notebooks are indexed. For example https://quantumai.google/cirq/google/best_practices, https://quantumai.google/cirq/google/qubit-placement, ..etc

File: docs/google/internal_gates.ipynb (hunk 2 above)
Comment (NoureldinYosri) [reply docs/google/internal_gates.ipynb]: done

**Pull request merged**